explainer/Explainer.md

```diff
@@ -1,9 +1,369 @@
+be difficult to justify.
+
+### IDs
+The Thing Description specification includes optional IDs which are mapped onto
+the RDF @id.  However, to support SPARQL queries, an @id is necessary in the 
+information model.  Therefore
+when an "anonymous" TD without an ID is registered with a Thing Description Directory,
+a temporary ID is created that is local to the directory.
+
+### Everything is a Thing
+Directories maintain lists of Thing Descriptions but are also Things 
+with their own Thing Description.
+
+The reason for this is the two-phase introduction/exploration process.
+There are introduction mechanisms that can only return a URL but cannot
+associate a "type" with that URL. A client would then not know whether the URL is pointing
+at a Thing Description or a Thing Description Directory.  We resolved this by deciding that
+URLs returned by introduction mechanisms always point at Thing Descriptions,
+but Thing Descriptions for Thing Description Directories have a special @type value.
+
+### Static Directories
+The only required affordance in a Thing Description Directory is "things",
+which lists all the contents of the directory (with support for pagination, however,
+to deal with long responses).  The ability to register, update, and delete
+Thing Descriptions in a directory is optional.
+
+The reason for this is to support the case
+of a small device that hosts multiple Things. For example, there exist
+"multisensors" that host multiple independent sensors for temperature, humidity,
+illumination, vibration, etc.  It is convenient for automation rules if
+each of these is modelled as a separate Thing. However, for self-description,
+the introduction URL needs to point at a single Thing Description.  We considered
+adding a Thing Descriptions with links representing items in a collection, 
+but settled on allowing small Thing Description Directories with static contents
+for this use case. 
+
+### Thing Links
+We wanted to support "federated" directory services where a query can be
+distributed over a set of directories.  SPARQL supports this feature but
+requires the set of federated directories to be given explictly in a query.
+For information modelling reasons, we wanted directories to also hold only
+a set of Thing Descriptions.  In order for directories to refer to other 
+directories for the purposes of federation, we support a special type of 
+Thing Description, a Thing Link, whose purpose is to refer to another
+Thing Description at another location.  This includes Thing Descriptions of
+directories, which are a kind of Thing.  Then federated searches can be set
+up by following these Thing Links.  We did not, however, make recursive searches
+automatic on the server side to avoid accidental or malicious exponential
+resource consumption.  Instead the client needs to explictly request searches
+from linked directories, and also has to have access rights to such linked
+directories.
```

Comment:
Might be useful to cite web crawlers here? the behavior explained above looks a lot like what a web crawler would do.

Reply:
might be helpful, suggest a change with a PR if you'd like.